ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/WatchedImages/WatchedImagesModal.tsx

```diff
@@ -21,53 +33,122 @@ function WatchedImagesModal({
     onClose,
     onWatchedImagesChange,
 }: WatchedImagesModalProps) {
-    const { data, mutate, isSuccess, isLoading, isError, error, reset } = useRestMutation(
-        (name: string) => watchImage(name),
-        { onSuccess: onWatchedImagesChange }
-    );
+    const watchedImagesFn = useCallback(getWatchedImages, []);
+    const currentWatchedImagesRequest = useRestQuery(watchedImagesFn);
+
+    const watchImageMutation = useRestMutation((name: string) => watchImage(name), {
+        onSuccess: onWatchedImagesChange,
+    });
+
+    const unwatchImageMutation = useRestMutation((name: string) => unwatchImage(name), {
+        onSuccess: () => {
+            onWatchedImagesChange();
+            currentWatchedImagesRequest.refetch();
+        },
+    });
 
     function onCloseModal() {
         onClose();
-        reset();
+        watchImageMutation.reset();
+        unwatchImageMutation.reset();
     }
 
+    const watchedImages = currentWatchedImagesRequest.data ?? [];
+
     return (
         <Modal
             title="Manage watched images"
             isOpen={isOpen}
             onClose={onCloseModal}
             variant="medium"
-            showClose={!isLoading}
-            onEscapePress={isLoading ? noop : onCloseModal}
+            showClose={false}
```

Comment:
As mentioned during our Friday sync, we'll always hide this button since an explicit "Close" button is available.